src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -142,17 +140,10 @@ void *frontendservicesesion(void *dummyPt) {
     bool loop_exit = false;
     int failCnt = 0;
     while (!loop_exit) {
-        string line = Utils::read_str_wrapper(connFd, data, FRONTEND_DATA_LENGTH, true);
+        std::string line = JasmineGraphFrontEndCommon::readAndProcessInput(connFd, data, failCnt);
         if (line.empty()) {
-            failCnt++;
-            if (failCnt > 4) {
-                break;
-            }
-            sleep(1);
```

Comment:
If you remove `sleep(1)` this code will start spinning infinitely. Why did you remove `sleep(1)`?

Reply:
i didnt remove the code segment, this segment is duplicated therefore i move this functionality to common file
```c++
JasmineGraphFrontEndCommon::readAndProcessInput(connFd, data, failCnt)
```

Reply:
Ok